sklearn/utils/_mask.py

```diff
@@ -1,21 +1,10 @@
 import numpy as np
 
-from . import is_scalar_nan
-from .fixes import _object_dtype_isnan
-
 
 def _get_mask(X, value_to_mask):
     """Compute the boolean mask X == missing_values."""
-    if is_scalar_nan(value_to_mask):
-        if X.dtype.kind == "f":
-            return np.isnan(X)
-        elif X.dtype.kind in ("i", "u"):
-            # can't have NaNs in integer array.
-            return np.zeros(X.shape, dtype=bool)
-        else:
-            # np.isnan does not work on object dtypes.
-            return _object_dtype_isnan(X)
+    if X.dtype.kind in ("i", "u"):
+        # can't have NaNs in integer array.
+        return np.zeros(X.shape, dtype=bool)
     else:
-        # X == value_to_mask with object dtypes does not always perform
-        # element-wise for old versions of numpy
-        return np.equal(X, value_to_mask)
+        return np.isnan(X)
```

Comment:
You are removing to much stuff. The missing value is not necessarily `nan`. Removing `object_dtype_isnan` should give something like
```py
if is_scalar_nan(value_to_mask):
    if X.dtype.kind == "f":
        return np.isnan(X)
    elif X.dtype.kind in ("i", "u"):
        # can't have NaNs in integer array.
        return np.zeros(X.shape, dtype=bool)
    else:
        # np.isnan does not work on object dtypes.
        return X != X
else:
    return np.equal(X, value_to_mask)
```